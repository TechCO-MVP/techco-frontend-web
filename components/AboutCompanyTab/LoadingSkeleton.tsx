import { Skeleton } from "@/components/ui/skeleton";

export default function LoadingSkeleton() {
  return (
    <div className="mx-auto w-full space-y-8 p-6">
      {/* Header Section */}
      <div className="space-y-2">
        <Skeleton className="h-8 w-3/4 bg-gray-300" />
        <Skeleton className="h-4 w-full bg-gray-300" />
      </div>

      {/* Company Profile Section */}
      <div className="flex items-start space-x-4">
        <Skeleton className="h-16 w-16 rounded-full bg-gray-300" />
        <div className="space-y-2">
          <Skeleton className="h-6 w-48 bg-gray-300" />
          <Skeleton className="h-4 w-32 bg-gray-300" />
        </div>
      </div>

      {/* Form Fields */}
      <div className="space-y-6">
        {/* Description Field */}
        <div className="space-y-2">
          <Skeleton className="h-4 w-32 bg-gray-300" />
          <Skeleton className="h-32 w-full bg-gray-300" />
        </div>

        {/* Website URL Field */}
        <div className="space-y-2">
          <Skeleton className="h-4 w-24 bg-gray-300" />
          <Skeleton className="h-10 w-full bg-gray-300" />
        </div>

        {/* LinkedIn Profile Field */}
        <div className="space-y-2">
          <Skeleton className="h-4 w-28 bg-gray-300" />
          <Skeleton className="h-10 w-full bg-gray-300" />
        </div>

        {/* Submit Button */}
        <Skeleton className="h-10 w-40 bg-gray-300" />
      </div>
    </div>
  );
}
