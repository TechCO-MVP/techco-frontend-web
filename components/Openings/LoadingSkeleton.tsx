import { Skeleton } from "../ui/skeleton";

export default function LoadingSkeleton() {
  return (
    <div className="flex h-full w-full flex-col px-8 py-6">
      <div className="space-y-6">
        <div className="flex items-center justify-between">
          <div className="space-y-2">
            <Skeleton className="h-8 w-96 bg-gray-300" />
            <Skeleton className="h-4 w-64 bg-gray-300" />
          </div>
          <Skeleton className="h-10 w-32 bg-gray-300" />{" "}
        </div>

        <div className="flex items-center justify-between">
          <Skeleton className="h-10 w-64 bg-gray-300" />
          <div className="flex gap-2">
            <Skeleton className="h-10 w-24 bg-gray-300" />
            <Skeleton className="h-10 w-24 bg-gray-300" />
          </div>
        </div>

        <div className="space-y-4">
          <div className="flex items-center gap-4 border-b p-4">
            {[
              "Nombre",
              "Correo electrÃ³nico",
              "Cargo",
              "Rol asignado",
              "Estado",
              "Candidatos",
              "Stakeholders",
              "prioridad",
              "",
            ].map((_, i) => (
              <Skeleton
                key={i}
                className="h-4"
                style={{ width: i === 5 ? "40px" : "160px" }}
              />
            ))}
          </div>

          {[...Array(3)].map((_, i) => (
            <div key={i} className="flex items-center gap-4 border-b p-4">
              <Skeleton className="h-4 w-40 bg-gray-300" />
              <Skeleton className="h-4 w-40 bg-gray-300" />
              <Skeleton className="h-4 w-40 bg-gray-300" />
              <Skeleton className="h-4 w-40 bg-gray-300" />
              <Skeleton className="h-4 w-40 bg-gray-300" />
              <Skeleton className="h-4 w-40 bg-gray-300" />
              <Skeleton className="h-4 w-40 bg-gray-300" />
              <Skeleton className="h-6 w-16 rounded-full bg-gray-300" />{" "}
              <Skeleton className="ml-auto h-4 w-4 bg-gray-300" />{" "}
            </div>
          ))}
        </div>
      </div>
    </div>
  );
}
