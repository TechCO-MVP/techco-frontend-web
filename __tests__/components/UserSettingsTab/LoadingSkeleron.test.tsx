import { render } from "@testing-library/react";
import LoadingSkeleton from "@/components/UserSettingsTab/LoadingSkeleton";
import { describe, it, expect } from "vitest";

describe("LoadingSkeleton", () => {
  it("renders correctly", () => {
    const { container } = render(<LoadingSkeleton />);
    expect(container).toMatchSnapshot();
  });
});
