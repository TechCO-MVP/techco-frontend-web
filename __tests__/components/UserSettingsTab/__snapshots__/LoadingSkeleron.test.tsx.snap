// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSkeleton > renders correctly 1`] = `
<div>
  <div
    class="flex h-full w-full flex-col px-8 py-6"
  >
    <div
      class="space-y-6"
    >
      <div
        class="flex items-center justify-between"
      >
        <div
          class="space-y-2"
        >
          <div
            class="animate-pulse rounded-md h-8 w-96 bg-gray-300"
          />
          <div
            class="animate-pulse rounded-md h-4 w-64 bg-gray-300"
          />
        </div>
        <div
          class="animate-pulse rounded-md h-10 w-32 bg-gray-300"
        />
         
      </div>
      <div
        class="flex items-center justify-between"
      >
        <div
          class="animate-pulse rounded-md h-10 w-64 bg-gray-300"
        />
         
        <div
          class="flex gap-2"
        >
          <div
            class="animate-pulse rounded-md h-10 w-24 bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md h-10 w-24 bg-gray-300"
          />
           
        </div>
      </div>
      <div
        class="space-y-4"
      >
        <div
          class="flex items-center gap-4 border-b p-4"
        >
          <div
            class="animate-pulse rounded-md bg-muted h-4"
            style="width: 160px;"
          />
          <div
            class="animate-pulse rounded-md bg-muted h-4"
            style="width: 160px;"
          />
          <div
            class="animate-pulse rounded-md bg-muted h-4"
            style="width: 160px;"
          />
          <div
            class="animate-pulse rounded-md bg-muted h-4"
            style="width: 160px;"
          />
          <div
            class="animate-pulse rounded-md bg-muted h-4"
            style="width: 160px;"
          />
          <div
            class="animate-pulse rounded-md bg-muted h-4"
            style="width: 40px;"
          />
        </div>
        <div
          class="flex items-center gap-4 border-b p-4"
        >
          <div
            class="animate-pulse rounded-md h-4 w-40 bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md h-4 w-40 bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md h-4 w-40 bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md h-4 w-40 bg-gray-300"
          />
           
          <div
            class="animate-pulse h-6 w-16 rounded-full bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md ml-auto h-4 w-4 bg-gray-300"
          />
           
        </div>
        <div
          class="flex items-center gap-4 border-b p-4"
        >
          <div
            class="animate-pulse rounded-md h-4 w-40 bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md h-4 w-40 bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md h-4 w-40 bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md h-4 w-40 bg-gray-300"
          />
           
          <div
            class="animate-pulse h-6 w-16 rounded-full bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md ml-auto h-4 w-4 bg-gray-300"
          />
           
        </div>
        <div
          class="flex items-center gap-4 border-b p-4"
        >
          <div
            class="animate-pulse rounded-md h-4 w-40 bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md h-4 w-40 bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md h-4 w-40 bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md h-4 w-40 bg-gray-300"
          />
           
          <div
            class="animate-pulse h-6 w-16 rounded-full bg-gray-300"
          />
           
          <div
            class="animate-pulse rounded-md ml-auto h-4 w-4 bg-gray-300"
          />
           
        </div>
      </div>
    </div>
  </div>
</div>
`;
